crates/criticalup-core/src/state.rs

```diff
@@ -374,6 +376,34 @@ impl StateInstallation {
     }
 }
 
+#[derive(Default)]
```

Comment:
I'm not sure `EnvVars::default()` resulting in a struct purely made of `None` values is what a programmer expects. Looking at `EnvVars` as a type name I would assume that `EnvVars::default()` would pull from the relevant environment. So I'd kind of assume it'd do the same thing as `EnvVars::read()`. As for an *actually* empty struct, I'd assume there would be an `EnvVars::empty()` I could call if I really needed (for testing etc).

Reply:
I think one of my issues was that I need to return an error when the string is not Unicode.